### PR TITLE
Scope MentorMembershipPolicy destroy? to school

### PR DESCRIPTION
**Pull request by gms-gs**
We need to scope to the school. So to active claims of a school can't have that mentor. Not all active claims.

### CODE REVIEW

**Review by CatalinVoineag — APPROVED** [review]
Nice, works as expected

File: app/controllers/claims/schools/mentors_controller.rb [hunk 1]
```diff
@@ -42,10 +45,18 @@ def set_mentor
     @mentor = @school.mentors.find(params.require(:id))
   end
 
+  def set_mentor_membership
+    @mentor_membership = @mentor.mentor_memberships.find_by!(school: @school)
+  end
```
Comment (Nitemaeric) [inline app/controllers/claims/schools/mentors_controller.rb:50]: This logic is duplicated on line 36. We should remove line 36 and use this reference instead.

File: app/controllers/claims/schools/mentors_controller.rb [hunk 2]
```diff
@@ -42,10 +45,18 @@ def set_mentor
...
+
   def authorize_mentor
     authorize @mentor || Claims::Mentor
   end
 
+  def authorize_mentor_membership
+    authorize @mentor_membership || Claims::MentorMembership
```
Comment (Nitemaeric) [inline app/controllers/claims/schools/mentors_controller.rb:57]: The `|| Claims::MentorMembership` is only needed when we want to authorise collections of mentor memberships. We don't need to use this for this case.

```ruby
authorize @mentor_membership
```

is sufficient.

File: app/controllers/claims/support/schools/mentors_controller.rb
Comment (Nitemaeric) [inline app/controllers/claims/support/schools/mentors_controller.rb:1]: Same comments as previous file.

**Pull request merged**